core/src/test/scala/org/apache/spark/shuffle/HostLocalShuffleFetchSuite.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.matchers.should.Matchers._
+
+import org.apache.spark.{LocalSparkContext, SparkConf, SparkContext, SparkEnv, SparkFunSuite, TestUtils}
+import org.apache.spark.internal.config._
+import org.apache.spark.network.netty.NettyBlockTransferService
+
+/**
+ * This test suite is used to test host local shuffle reading with external shuffle service disabled
+ */
+class HostLocalShuffleFetchSuite extends SparkFunSuite with Matchers with LocalSparkContext {
```

Comment:
Thank you for adding this.

Reply:
I've combined the two host-local shuffle reading tests(with external shuffle service enabled or disabled) into the [HostLocalShuffleReadingSuite](https://github.com/apache/spark/blob/6b97be552b1d5a78f476c4a97d794c567222d9bf/core/src/test/scala/org/apache/spark/shuffle/HostLocalShuffleReadingSuite.scala). Thus, the host-local shuffle reading feature can be tested centralized. Does it look okay to you? @dongjoon-hyun @attilapiros

Reply:
I see value in this test but @dongjoon-hyun is right too (currently the two asserts can be done with some small unit tests too). 

To improve this I would make sure that really our feature worked under the hood. 
For example I would check the `ShuffleReadMetrics` as there should be no remote block fetching.

Reply:
Sorry, which two asserts are you referring to? Are these two asserts in the new test:

```scala
// Spark should read the shuffle data locally from the cached directories on the same host,
// so there's no remote fetching at all.
assert(localBytesRead.sum > 0)
assert(remoteBytesRead.sum === 0)
```
If they are, I actually think that checking `localBytesRead`/`remoteBytesRead` is equal to `localBlocksFetched`/`remoteBlocksFetched` here.

Reply:
I was referring to the asserts mentioned by @dongjoon-hyun:
- https://github.com/apache/spark/pull/28911#pullrequestreview-478159228
- https://github.com/apache/spark/pull/28911#pullrequestreview-478159246

Those could be in UTs.

Reply:
hmm...I'm not sure whether @dongjoon-hyun was mentioning those two asserts. But I can try to make them as separate unit tests.

Reply:
I think you can keep those as it is just extend this integration test with more assert to make sure the feature really works (through all the components). This is why I suggested to check the metrics too.

Reply:
Thanks. I already added the checking for metrics. Let me try to add other assertions to catch the expected behavior.